### ISSUE TITLE
Deploy Mantl platform breaks on Distributive role (install distributive, checklists for role)

### ISSUE THREAD

**Issue by d-fault** [mantl/mantl#2022]
This url [https://dl.bintray.com/shippedrepos/mantl-1.3/](url) can no longer be used for package downloads. It's forbidden. Does anyone know what's happening, how can i fix that

**Comment by Theaxiom** [comment]
It seems that Cisco closed all the repos for Mantl from bintray. You can get the updated packages here: https://github.com/mantl/mantl/wiki/Notes

So you would change all the corresponding references to the following:

- https://bintray.com/mantl/rpm-packages
- https://dl.bintray.com/mantl/rpm-packages
- https://dl.bintray.com/mantl/rpm-packages/repodata/repomd.xml

**Comment by Theaxiom** [comment]
Looks like we need to change all these references in the repo, who's up for the task?

**Comment by shaunstoltz** [comment]
Hmm, seems this project is useless now, nothing works... again. Disaster.

**Comment by Theaxiom** [comment]
@shaunstoltz We are working to re-write Mantl from the ground up: 

- https://github.com/mantl/mantl/issues/1947
- https://github.com/Dynamictivity/endeavor

**Comment by Theaxiom** [comment]
I had already duplicated and updated the existing packages to a new bintray, long before Cisco disabled them.

**Comment by Cryptophobia** [comment]
We are still missing the `distributive` packages with the above changes. Should we use `https://bintray.com/asteris/mantl-rpm` until we figure out how to proceed?

**Comment by Theaxiom** [comment]
We should copy the missing packages to the new repo.

Thanks,
Travis Rowland
http://travisrowland.com
http://linkedin.com/in/travisrowland
http://about.me/travisrowland

-----------------------------------------------

This e-mail is privileged and confidential.  Do not forward, copy or print
without authorization.  If you are not the intended recipient of this
E-mail, you are hereby notified that any dissemination, distribution or
copying of this message is strictly prohibited. If misdirected, please
delete and notify the sender by e-mail at travis.rowland@gmail.com.

CAUTION: E-mail sent through the internet is not secure and could be
intercepted by a third party. For your protection, avoid sending
identifying information, such as account numbers or social security numbers.

**Comment by Theaxiom** [comment]
- `etcd-3.0.15`
- `mesos-0.28.2`?

(not sure if 1 & 2 would be required on new install):
1.  Install `deltarpm` to control nodes.
2. Invoke `sudo yum clean all`
2. Change mesos version from `0.28.2` to `0.28.0` to match available package on this server.

**Comment by Cryptophobia** [comment]
I am getting the below SSH errors when I try to provision a vagrant mantl cluster:

```
GATHERING FACTS *************************************************************** 
ok: [control-01]
fatal: [kubeworker-001] => SSH Error: ssh: connect to host 172.20.10.151 port 22: No route to host
    while connecting to 172.20.10.151:22
It is sometimes useful to re-run the command using -vvvv, which prints SSH debug output to help diagnose the issue.
fatal: [edge-001] => SSH Error: ssh: connect to host 172.20.10.251 port 22: No route to host
    while connecting to 172.20.10.251:22
It is sometimes useful to re-run the command using -vvvv, which prints SSH debug output to help diagnose the issue.
fatal: [worker-001] => SSH Error: ssh: connect to host 172.20.10.201 port 22: No route to host
    while connecting to 172.20.10.201:22
It is sometimes useful to re-run the command using -vvvv, which prints SSH debug output to help diagnose the issue.
```

Any ideas? All the routes and networks in VirtualBox Manager and Vagrant seem to be correctly configured.

**Comment by Theaxiom** [comment]
@Cryptophobia Shouldn't that be a new issue?

**Comment by Cryptophobia** [comment]
@Theaxiom , Yes I thought maybe it was simple. But let me make a new issue about it.

Did you test latest master on Vagrant?

**Comment by Cryptophobia** [comment]
Opened a new issue here: https://github.com/mantl/mantl/issues/2023

**Comment by elordahl** [comment]
vault-0.6.0 (although i see 1.7.0)
marathon-1.1.1

**Comment by elordahl** [comment]
hey @Theaxiom, do you have an ETA on the missing packages?

**Comment by Theaxiom** [comment]
@elordahl I will work to get those uploaded today.

**Comment by Theaxiom** [comment]
I added 4 new packages to the mantl/rpm-packages YUM repo:
- https://bintray.com/mantl/rpm-packages/distributive/v0.2.5
- https://bintray.com/mantl/rpm-packages/docker-cleanup/0.0.1-1
- https://bintray.com/mantl/rpm-packages/mesos/0.28.0-4
- https://bintray.com/mantl/rpm-packages/marathon/1.0 (still uploading, will update when complete)

**Comment by elordahl** [comment]
Thanks!  Do you intend to upload any of the v1.3 packages?

**Comment by Theaxiom** [comment]
@elordahl I don't plan to, we are only supporting 2.0 right now as LTS while we re-write Mantl from scratch. There is way too much bit-rot here to build upon and we need to break away from maintaining a k8s fork. You can check out the progress on the new Mantl here: https://github.com/cloud-endeavor/endeavor

**Comment by Theaxiom** [comment]
All uploads are completed.

**Comment by Theaxiom** [comment]
@elordahl I apologize for the confusion, I said 2.0 above but never mind. Yes, all 1.3 packages should be available at that repository location.

**Comment by Theaxiom** [comment]
If there are still any 1.3 packages missing, please let me know and I will upload them.

**Comment by elordahl** [comment]
I appreciate the support----Looking forward to the new mantl.  Am i looking in the wrong location?  I still dont see the following, which come from the 1.3.0 tag:

`etcd-3.0.15` (i'm manually resolving this)
`mesos-0.28.2` - i only see 0.28.0, which is not a huge deal, but broken w/o modification.
`marathon-1.1.1` - i only see 1.0.0-RC
`vault-0.6.0` - i do see 0.7.0, which should be fine.

**Comment by Theaxiom** [comment]
@elordahl I uploaded the latest versions of the packages, if you require specific versions please list them and I will get them uploaded.

I didn't see etcd-3.0.15 on asteris, I will check for it on shipped.

**Comment by Theaxiom** [comment]
@elordahl I need to slow down, sorry. I see the versions you listed, I will get them uploaded. 👍

**Comment by Theaxiom** [comment]
Looks like `etcd-3.0.15` wasn't coming from shippedrepos or asteris. Where are you resolving it from @elordahl ?

**Comment by Theaxiom** [comment]
I have no way to obtain `mesos-0.28.2` as it is forbidden for downloading: https://shippedrepos.bintray.com/mantl-1.3/mesos-0.28.2-1.x86_64.rpm

Does anyone have this RPM available?

**Comment by Theaxiom** [comment]
Same issue for `marathon-1.1.1` as with `mesos-0.28.2`, forbidden download: https://shippedrepos.bintray.com/mantl-1.3/marathon-1.1.1-1.x86_64.rpm

**Comment by Theaxiom** [comment]
It looks like we will need some help to obtain those packages.

**Comment by elordahl** [comment]
i found etcd on an rpmfind ftp server 😩 `ftp://fr2.rpmfind.net/linux/centos/7.3.1611/extras/x86_64/Packages/etcd-3.0.15-1.el7.x86_64.rpm`

**Comment by Cryptophobia** [comment]
This is a mess. Do we know who we can ask about those rpm packages?

**Comment by elordahl** [comment]
Any luck? Any updates?  I rebuild environments frequently, so these missing packages are forcing me to change my infrastructure (and regret not configuring a proxy/cache).

**Comment by Theaxiom** [comment]
@elordahl Unfortunately I do not have these packages (`marathon-1.1.1` and `mesos-0.28.2`), however they may be able to be reproduced using our packing repos. It uses vagrant to perform an automated process for packaging. I will work over the weekend in an attempt to reproduce these packages. Thank you for your patience and understanding! 👍

**Comment by elordahl** [comment]
@Theaxiom Thanks much!

**Comment by Cryptophobia** [comment]
@Theaxiom Can we ping someone from Asteris, maybe they have the packages and can upload some of those packages sooner?

@stevendborrelli ?

**Comment by Theaxiom** [comment]
I am sorry I didn't get to this yet, I was up all weekend working on my day job. I have not forgotten about this, if someone else wants to take a whack at this here are the package repos:
- https://github.com/mantl/mantl-packaging
- https://github.com/mantl/mesos-packaging

**Comment by stevendborrelli** [comment]
Taking a look right now. @Theaxiom @Cryptophobia @elordahl

**Comment by Theaxiom** [comment]
Thanks @stevendborrelli !

**Comment by stevendborrelli** [comment]
Can folks check now? https://dl.bintray.com/shippedrepos/mantl-1.3/

I don't know what happened. We may want to move these package somewhere else.

**Comment by Theaxiom** [comment]
@stevendborrelli Forbidden: https://dl.bintray.com/shippedrepos/mantl-1.3/marathon-1.1.1-1.x86_64.rpm

I have already created a new repo on bintray, but I am missing those 2 packages.

**Comment by stevendborrelli** [comment]
I think I see the issue, looks like some automated push of docker containers (for other cisco projects) has caused it to hit rate limits. Let me see what I can do.

**Comment by Theaxiom** [comment]
Thanks again @stevendborrelli 👍

**Comment by stevendborrelli** [comment]
I have a ticket open with JFrog...

**Comment by stevendborrelli** [comment]
I talked to JFrog. The account that paid for this repo stopped paying, and after a huge amount of emails, JFrog got permission to shut it down. They might be able to get a couple of the files for us to host somewhere else. 

@Theaxiom can you give me a list of packages that we need?

**Comment by Theaxiom** [comment]
@stevendborrelli Those are the only 2 we are missing. Thanks!

**Comment by elordahl** [comment]
Hey all, I had to bail on Mantl to remove some bottlenecks and meet deadlines, but I appreciate the support nonetheless.  I'm looking forward to the next gen.  Thanks again!

**Comment by Jamie-BitFlight** [comment]
Is there a way to build mantl any more?

If so, which files need modifying to use the new packages above for a new deployment?

**Comment by stevendborrelli** [comment]
I have not heard back from JFrog. 

Mantl packaging is here: https://github.com/mantl/mantl-packaging. There is a Virtualbox vm you can spin up to build all the packages.

**Comment by Theaxiom** [comment]
@Jamie-BitFlight Yes, we already have rebuilt all the packages and pushed them to a new repo on Bintray several months ago. The problem is it seems some package requirements involve lesser versions of some packages. We will work to fill this gap and get Mantl back to an installable status.

Thank you @stevendborrelli

**Comment by Theaxiom** [comment]
We are going to have to figure out if the newer versions of these softwares can indeed be used, or as @Jamie-BitFlight stated, what needs to be modified overall to make everything mesh.

**Comment by stevendborrelli** [comment]
The mesos-packaging repo https://github.com/mantl/mesos-packaging is the one that was used to build the packages. I was able to re-create the marathon-1.1.1 package, but the mesos one needs to be run on a dedicated linux box. (My virtualbox VM ended up crashing my mac during the build).

**Comment by Cryptophobia** [comment]
@Theaxiom , can we just upload the packages that Steven made and see if the deployment/provisioning works now?

**Comment by Theaxiom** [comment]
@Cryptophobia I don't have the packages.
